include/seahorn/Expr/ExprOpStruct.hh

```diff
@@ -138,7 +138,7 @@ inline Expr push_ite_struct(Expr c, Expr lhs, Expr rhs) {
 
   llvm::SmallVector<Expr, 8> vals;
   for (unsigned i = 0, sz = lhs->arity(); i < sz; ++i) {
-    vals.push_back(boolop::lite(c, lhs->arg(i), rhs->arg(i)));
+    vals.push_back(bind::lite(c, lhs->arg(i), rhs->arg(i)));
```

Comment:
this is an ok fix for current problem, but I am afraid that we might not be handling all combinations of structs and lambdas. For instance, `bind::lite` does not call `push_ite_struct`, so creating an `ite` over a `lambda` that returns a `struct` will not be handled correctly.